apps/aecore/src/aec_conductor.erl

```diff
@@ -1330,6 +1340,42 @@ handle_key_block_candidate_reply({{error, Reason}, _}, State) ->
     epoch_mining:error("Creation of key block candidate failed: ~p", [Reason]),
     create_key_block_candidate(State).
 
+hc_create_block_fun(ConsensusModule, TopHash) ->
+    fun() ->
+        case get_next_beneficiary(ConsensusModule, TopHash) of
+            {ok, Leader} ->
+                  epoch_mining:debug("Got leader, calling hc_create_block", []),
+                  {hc_create_block(ConsensusModule, TopHash, Leader), TopHash};
+            {error, _} = Err ->
+                {Err, TopHash}
+        end
+    end.
+
+hc_create_block(ConsensusModule, TopHash0, Leader) ->
+    TopHash = hc_create_microblock(ConsensusModule, TopHash0, Leader),
+    Res = aec_block_key_candidate:create(TopHash, Leader, Leader),
+    Res.
+
+hc_create_microblock(ConsensusModule, TopHash, Leader) ->
+    case aec_block_micro_candidate:create(TopHash) of
+        {ok, MBlock, _MBlockInfo} ->
+            case aec_blocks:txs(MBlock) of
+                [] ->
+                    epoch_mining:debug("Empty micro-block, top is still: ~p", [TopHash]),
+                    TopHash;
+                [_ | _ ] ->
+                    SignModule = ConsensusModule:get_sign_module(),
+                    {ok, SignedMBlock} = SignModule:sign_micro_block(MBlock, Leader),
+                    add_signed_block(SignedMBlock),
+                    {ok, MBHash} = aec_blocks:hash_internal_representation(SignedMBlock),
+                    epoch_mining:debug("New micro-block added, top is: ~p", [MBHash]),
+                    MBHash
+            end;
+        _ ->
```

Comment:
Tracing the code a bit and it seems this can only happen in case where something is actually wrong. Possibly log an info here instead of debug.

Reply:
Makes sense!